servicetalk-concurrent-internal/src/main/java/io/servicetalk/concurrent/internal/EmptySubscriptions.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright © 2018, 2021 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.concurrent.internal;
+
+import io.servicetalk.concurrent.PublisherSource.Subscriber;
+import io.servicetalk.concurrent.PublisherSource.Subscription;
+
+import static io.servicetalk.concurrent.internal.SubscriberUtils.isRequestNValid;
+import static io.servicetalk.concurrent.internal.SubscriberUtils.newExceptionForInvalidRequestN;
+
+/**
+ * Utility methods for {@link Subscription}s which don't deliver any data.
+ */
+public final class EmptySubscriptions {
+    public static final Subscription EMPTY_SUBSCRIPTION = newEmptySubscription(true);
+    public static final Subscription EMPTY_SUBSCRIPTION_NO_THROW = newEmptySubscription(false);
+
+    private EmptySubscriptions() {
+    }
+
+    /**
+     * Create an empty {@link Subscription} that will propagate an error to a {@link Subscriber} upon invalid demand.
+     * @param subscriber The subscriber to propagate
+     * @param <T> The type of {@link Subscriber}.
+     * @return An empty {@link Subscription} that will propagate an error to a {@link Subscriber} upon invalid demand.
+     */
+    public static <T> Subscription newEmptySubscription(Subscriber<T> subscriber) {
+        return new Subscription() {
+            private boolean terminated;
+            @Override
+            public void request(final long n) {
+                if (!terminated && !isRequestNValid(n)) {
+                    terminated = true;
+                    subscriber.onError(newExceptionForInvalidRequestN(n));
+                }
+            }
+
+            @Override
+            public void cancel() {
+            }
+        };
+    }
+
+    /**
+     * Create an empty {@link Subscription} that will optionally validate invalid demand.
+     * @param throwOnInvalidDemand {@code true} to throw on invalid {@link Subscription#request(long)} demand.
+     * {@code false} will not validate demand.
+     * @return A {@link Subscription} that will optionally validate invalid demand.
+     */
+    public static Subscription newEmptySubscription(boolean throwOnInvalidDemand) {
```

Comment:
If we need to use this method in the future for new use-cases what is the rationale for choosing between `true` and `false`? Consider clarifying that in the javadoc, it will help new team members and OSS folks.

Reply:
I'll make this internal since all usages are through the `static final` variable.